### ISSUE TITLE
[feature]  new __repr__ function

### ISSUE THREAD

**Issue by lewisacidic** [lewisacidic/jagged-array#16]
Repr will be broken when we address #14 as the shapes will get quite long vertically.

We should probably realign the repr to look like pydata/sparse for consistency.

**Comment by lewisacidic** [comment]
from #1, we should be numpy-like and not output the dtype if it is the standard float/int type.

**Comment by lewisacidic** [comment]
We should emulate numpy, should be easy to get a passable repr with [array2string](https://docs.scipy.org/doc/numpy/reference/generated/numpy.array2string.html#numpy.array2string)

Something like:

```python
>>> JaggedArray(np.array(8), (3, (3, 2, 3), dtype='i4')
JaggedArray([[0, 1, 2],
             [3, 4],
             [5, 6, 7]], dtype=int32)
```

**Comment by lewisacidic** [comment]
Implemented in e96c215.
